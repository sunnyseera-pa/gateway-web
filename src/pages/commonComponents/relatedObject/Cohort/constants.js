const tab_url = '/search?search=&tab=Cohorts';
export const cohort = {
	TAB: 'Cohort',
	FC: {
		parentKey: 'cohortinclusionexclusion',
		filter: 'cohortFilterCriteriaSelected',
		tagType: 'tag',
		url: `${tab_url}&cohortinclusionexclusion=`,
	},
};
