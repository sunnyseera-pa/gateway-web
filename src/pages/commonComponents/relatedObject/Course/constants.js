const tab_url = '/search?search=&tab=Courses';
export const course = {
	TAB: 'Course',
	AWARDS: {
		parentKey: 'courseaward',
		filter: 'courseaward',
		tagType: 'tag',
		url: `${tab_url}&courseaward=`,
	},
	DOMAINS: {
		parentKey: 'coursedomains',
		filter: 'courseDomainsSelected',
		tagType: 'tag',
		url: `${tab_url}&coursedomains=`,
	},
};
