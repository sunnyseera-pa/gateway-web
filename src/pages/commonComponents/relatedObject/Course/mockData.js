export default {
    _id: '5f96f3450d236c6cafdaa473',
    domains: ['test'],
    award: ['BSc'],
    courseOptions: [
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
    ],
    id: 3491286126459112,
    type: 'course',
    activeflag: 'active',
    title: 'Test A course',
    provider: 'Test A',
    description: 'Test A Test A Test A edited Test A',
    counter: 208,
    relatedresources: 3,
    updated: 1621465200000,
};
