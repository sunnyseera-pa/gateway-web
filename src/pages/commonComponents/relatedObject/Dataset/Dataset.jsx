/** @jsx jsx */
import { jsx } from '@emotion/react';
import React, { useState, useEffect, useCallback } from 'react';
import queryString from 'query-string';
import { Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { isEmpty, isNil } from 'lodash';
import googleAnalytics from '../../../../tracking';
import { stripMarkdown } from '../../../../utils/GeneralHelper.util';
import SVGIcon from '../../../../images/SVGIcon';
import RemoveButton from '../RemoveButton/RemoveButton';
import Description from '../Description/Description';
import Tag from '../Tag/Tag';
import * as styles from './Dataset.styles';
import '../../CommonComponents.scss';
import '../RelatedObject.scss';

const Dataset = ({
	data,
	activeLink,
	publisherLogo,
	onSearchPage,
	showRelationshipQuestion,
	isCohortDiscovery,
	updateOnFilterBadge,
	removeButton,
}) => {
	const [publisherDetails, setPublisherDetails] = useState({ name: '', label: '' });

	const getPublisherDetails = useCallback(() => {
		let publisher = { name: '', label: '', showShield: false };
		if (!isEmpty(data.datasetv2)) {
			let name = data.datasetv2.summary.publisher.name.toUpperCase();
			publisher.name = name;
			publisher.label = name;
			publisher.showShield = !isNil(data.datasetv2.summary.publisher.memberOf) ? true : false;
		} else {
			let name = data.datasetfields.publisher;
			let publisherName = name.includes('>') ? name.split(' > ')[1].toUpperCase() : name.toUpperCase();
			publisher.name = publisherName;
			publisher.label = publisherName;
			publisher.showShield = false;
		}
		return publisher;
	}, [data.datasetv2, data.datasetfields.publisher]);

	useEffect(() => {
		let publisherDetails = getPublisherDetails();
		setPublisherDetails(publisherDetails);
	}, [getPublisherDetails]);

	const getDescription = () => {
		if (!data.datasetfields.abstract || typeof data.datasetfields.abstract === 'undefined') {
			if (data.description) {
				return stripMarkdown(data.description, 255);
			}
		} else {
			return stripMarkdown(data.datasetfields.abstract);
		}
	};

	if (data.type === 'dataset' && data.activeflag === 'archive') {
		return (
			<Row className='noMargin pad-left-24'>
				<Col sm={10} lg={10} className='entity-deleted-edit gray800-14'>
					The dataset '{data.name}' has been deleted by the publisher
				</Col>
				<Col sm={2} lg={2}>
					<RemoveButton removeButtonHandler={removeButton} />
				</Col>
			</Row>
		);
	}

	const phenotypesSelected = queryString.parse(window.location.search).phenotypes
		? queryString.parse(window.location.search).phenotypes.split('::')
		: [];
	const searchTerm = queryString.parse(window.location.search).search ? queryString.parse(window.location.search).search : '';
	const phenotypesSearched = data.datasetfields.phenotypes.filter(phenotype => phenotype.name.toLowerCase() === searchTerm.toLowerCase());
	return (
		<>
			<Row data-testid='related-dataset-object' className='noMargin'>
				<Col sm={10} lg={10} className='pad-left-24'>
					{activeLink ? (
						<a
							onClick={() => {
								googleAnalytics.recordEvent('Datasets', 'Clicked on dataset to open', `Dataset name: ${data.name}`);
							}}
							className='purple-bold-16'
							css={styles.pointer}
							href={'/dataset/' + data.pid}
							data-testid='dataset-title'>
							{data.name}
						</a>
					) : (
						<span className='black-bold-16' data-testid='dataset-title'>
							{' '}
							{data.name}{' '}
						</span>
					)}
					<br />
					{publisherDetails.showShield && (
						<span>
							<SVGIcon name='shield' fill={'#475da7'} className='svg-16 mr-2' viewBox='0 0 16 16' />
						</span>
					)}
					<span
						className={activeLink ? 'gray800-14 underlined' : 'gray800-14'}
						css={styles.pointer}
						onClick={() =>
							updateOnFilterBadge('publisher', {
								label: publisherDetails.label,
								parentKey: 'publisher',
							})
						}
						data-testid={`publisher-${publisherDetails.name}`}>
						{' '}
						{publisherDetails.name}{' '}
					</span>
				</Col>
				<Col sm={2} lg={2} className='pad-right-24'>
					{!isEmpty(publisherLogo) && (
						<div className='datasetLogoCircle floatRight' css={styles.publisherLogoCSS(publisherLogo)} data-testid='publisher-logo' />
					)}
					{showRelationshipQuestion && <RemoveButton removeButtonHandler={removeButton} />}
				</Col>
				<Col sm={12} lg={12} className='pad-left-24 pad-right-24 pad-top-16'>
					<Tag
						tagName='Dataset'
						tagType='dataset'
						activeLink={false}
						onSearchPage={false}
						parentKey=''
						url='/search?search=&tab=Datasets'
						updateOnFilterBadgeHandler={updateOnFilterBadge}
						showTagType={false}>
						<SVGIcon name='dataseticon' fill={'#113328'} className='badgeSvg mr-2' viewBox='-2 -2 22 22' />
					</Tag>
					{isCohortDiscovery && (
						<Tag
							tagName='Cohort Discovery'
							tagType='project'
							activeLink={false}
							onSearchPage={false}
							parentKey=''
							url='/search?search=&tab=Datasets'
							updateOnFilterBadgeHandler={updateOnFilterBadge}
							showTagType={false}>
							<SVGIcon name='cohorticon' fill={'#472505'} className='badgeSvg mr-2' width='22' height='22' viewBox='0 0 10 10' />
						</Tag>
					)}

					{phenotypesSearched && phenotypesSearched.length > 0 && (
						<Tag
							key={`phenotypes-searched`}
							tagName={phenotypesSearched[0].name}
							tagType='phenotype'
							activeLink={activeLink}
							onSearchPage={onSearchPage}
							parentKey='phenotypes'
							url='/search?search=&tab=Datasets&phenotypes='
							updateOnFilterBadgeHandler={updateOnFilterBadge}
							showTagType={true}
						/>
					)}

					{phenotypesSelected &&
						phenotypesSelected.length > 0 &&
						phenotypesSelected.map((phenotype, index) => {
							if (data.datasetfields.phenotypes.find(phenotypeCheck => phenotypeCheck.name.toLowerCase() === phenotype.toLowerCase())) {
								return (
									<Tag
										key={`phenotypes-selected-${index}`}
										tagName={phenotype}
										tagType='phenotype'
										activeLink={activeLink}
										onSearchPage={onSearchPage}
										parentKey='phenotypes'
										url='/search?search=&tab=Datasets&phenotypes='
										updateOnFilterBadgeHandler={updateOnFilterBadge}
										showTagType={true}
									/>
								);
							} else {
								return null;
							}
						})}

					{data.tags.features &&
						data.tags.features.length > 0 &&
						data.tags.features.map((feature, index) => {
							return (
								<Tag
									key={`tag-${index}`}
									tagName={feature}
									tagType='tag'
									activeLink={activeLink}
									onSearchPage={onSearchPage}
									parentKey='datasetfeatures'
									url='/search?search=&tab=Datasets&datasetfeatures='
									updateOnFilterBadgeHandler={updateOnFilterBadge}
									showTagType={false}
								/>
							);
						})}
				</Col>
				{!showRelationshipQuestion && <Description type={data.type} description={getDescription()} />}
			</Row>
		</>
	);
};

Dataset.propTypes = {
	data: PropTypes.object.isRequired,
	activeLink: PropTypes.bool.isRequired,
	publisherLogo: PropTypes.string,
	showRelationshipQuestion: PropTypes.bool.isRequired,
	onSearchPage: PropTypes.bool.isRequired,
	isCohortDiscovery: PropTypes.bool.isRequired,
	updateOnFilterBadge: PropTypes.func.isRequired,
	removeButton: PropTypes.func.isRequired,
};

export default Dataset;
