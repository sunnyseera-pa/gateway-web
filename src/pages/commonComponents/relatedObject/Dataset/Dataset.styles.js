import { css } from '@emotion/react';

export const publisherLogoCSS = publisherLogo => css`
	background-image: url('${publisherLogo}');
	background-repeat: no-repeat;
	background-position: center;
	background-size: contain;
	background-origin: content-box;
`;
export const pointer = css`
	cursor: pointer;
`;
