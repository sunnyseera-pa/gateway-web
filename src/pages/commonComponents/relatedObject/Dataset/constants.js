const tab_url = '/search?search=&tab=Datasets';
export const dataset = {
	TAB: 'Dataset',
	PHENOTYPES: {
		parentKey: 'phenotypes',
		filter: 'phenotypes',
		tagType: 'phenotype',
		url: `${tab_url}&phenotypes=`,
	},
	FEATURES: {
		parentKey: 'datasetfeatures',
		filter: 'datasetfeatures',
		tagType: 'tag',
		url: `${tab_url}&datasetfeatures=`,
	},
};
