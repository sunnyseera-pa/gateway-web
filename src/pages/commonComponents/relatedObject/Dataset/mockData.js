export default {
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['COVID-19', 'TRACKER', 'SAIL', 'COVID', 'MOBILE APP', 'CORONAVIRUS', 'SYMPTOM', 'NCS', 'National Core Study'],
        topics: [],
    },
    datasetfields: {
        publisher: 'ALLIANCE > TestSAIL',
        geographicCoverage: ['Great Britain'],
        physicalSampleAvailability: ['NOT AVAILABLE'],
        abstract: 'This is a test dataset abstract',
        ageBand: '15-150',
        metadataquality: {
            quality_score: 89.48,
        },
        phenotypes: [],
    },
    programmingLanguage: [],
    pid: 'c1f4b16c-9dfa-48e5-94ee-f0aa58c270e4',
    id: 401490453939233,
    datasetid: '78ce070c-84db-4676-81c5-0c672f4e12e3',
    type: 'dataset',
    activeflag: 'active',
    name: 'COVID-19 Symptom Tracker Dataset',
    description: 'This is a test dataset description',
    license: 'https://covid.joinzoe.com/privacy-notice',
    datasetv2: {
        summary: {
            publisher: {
                name: 'sail',
                logo: '',
                memberOf: 'ALLIANCE',
            },
        },
    },
    counter: 955,
    isCohortDiscovery: false,
    persons: [
        {
            firstname: 'Rich',
            lastname: 'Example',
        },
    ],
    latestUpdate: '2021-05-04T14:14:49.389Z',
    relatedresources: 1,
};
