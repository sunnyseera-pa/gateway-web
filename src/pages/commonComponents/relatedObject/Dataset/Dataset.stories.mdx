import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Dataset from './Dataset';
import mockData from './mockData';

<Meta
	title='Related Resources'
	component={Dataset}
	args={{
		data: mockData,
		onSearchPage: false,
		isCohortDiscovery: false,
		publisherLogo: 'https://uatbeta.healthdatagateway.org/allianceMembers/NHS-Digital-logo_RGB-01.8952cf62.jpg',
		activeLink: false,
		showRelationshipQuestion: false,
	}}
/>

<Canvas>
	<Story name='Dataset'>
		{props => {
			const updateOnFilterBadge = () => {
				alert('updateOnFilterBadge');
			};
			const removeButton = () => {
				alert('removed');
			};
			return <Dataset {...props} updateOnFilterBadge={updateOnFilterBadge} removeButton={removeButton} />;
		}}
	</Story>
</Canvas>
