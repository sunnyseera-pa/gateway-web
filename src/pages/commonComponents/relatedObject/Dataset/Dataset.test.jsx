import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Dataset from './Dataset';
import mockData from './mockData';
import { dataset } from './constants';
const props = {
	data: { ...mockData },
	onSearchPage: false,
	isCohortDiscovery: false,
	publisherLogo: 'https://uatbeta.healthdatagateway.org/allianceMembers/NHS-Digital-logo_RGB-01.8952cf62.jpg',
	activeLink: false,
	showRelationshipQuestion: false,
	updateOnFilterBadge: jest.fn(),
	removeButton: jest.fn(),
};
let wrapper;
let handlerFn;

describe('Given the Dataset component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<Dataset {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then Dataset Title should be rendered with description', () => {
			expect(screen.getByTestId('dataset-title')).toHaveTextContent(props.data.name);
			expect(screen.getByTestId('dataset-description')).toHaveTextContent(props.data.datasetfields.abstract);
		});

		it('Then Dataset SVG Icon should be rendered', () => {
			expect(screen.getByTestId('dataseticon')).toBeTruthy();
		});

		it('Then Publisher Name should be rendered in UpperCase', () => {
			const publisher = screen.getByTestId(`publisher-${props.data.datasetv2.summary.publisher.name.toUpperCase()}`);
			expect(publisher).toBeTruthy();
			expect(publisher).toHaveTextContent(props.data.datasetv2.summary.publisher.name.toUpperCase());
		});
		it('Then Publisher Link should be rendered with shield SVG Icon', () => {
			expect(screen.getByTestId(`publisher-${props.data.datasetv2.summary.publisher.name.toUpperCase()}`)).toBeTruthy();
			expect(screen.getByTestId(`shield`)).toBeTruthy();
		});

		it('Then Publisher Name onclick updateOnFilterBadge should be called', () => {
			let updateOnFilterBadge = props.updateOnFilterBadge;
			fireEvent.click(screen.getByTestId(`publisher-${props.data.datasetv2.summary.publisher.name.toUpperCase()}`));
			expect(updateOnFilterBadge.mock.calls.length).toBe(1);
			expect(updateOnFilterBadge.mock.calls[0][0]).toEqual('publisher');
			expect(updateOnFilterBadge.mock.calls[0][1]).toEqual({
				label: props.data.datasetv2.summary.publisher.name.toUpperCase(),
				parentKey: 'publisher',
			});
		});

		it('Then Publisher Logo should be rendered', () => {
			expect(screen.getByTestId('publisher-logo')).toBeTruthy();
		});

		it('Then the Badge Tags/Features should be rendered without links', () => {
			props.data.tags.features.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
			});
		});
	});
	describe('And isCohortDiscovery is true', () => {
		it('Then the CohortDiscovery SVG icon should be rendered', () => {
			const { rerender } = wrapper;
			handlerFn = jest.fn();
			rerender(<Dataset {...props} isCohortDiscovery={true} />);
			expect(screen.getByTestId('cohorticon')).toBeTruthy();
		});
	});

	describe('And isCohortDiscovery is true', () => {
		it('Then the CohortDiscovery SVG icon should be rendered', () => {
			const { rerender } = wrapper;
			rerender(<Dataset {...props} isCohortDiscovery={true} />);
			expect(screen.getByTestId('cohorticon')).toBeTruthy();
		});
	});

	describe('And activeLink is true', () => {
		it('Then the Tilte should be clickable with a link', () => {
			const { rerender } = wrapper;
			rerender(<Dataset {...props} activeLink={true} />);
			expect(screen.getByTestId('dataset-title')).toHaveAttribute('href', `/dataset/${props.data.pid}`);
		});
		it('Then the Badge Tags/Features should be rendered with links', () => {
			props.data.tags.features.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.getByTestId(`badge-${value}-link`)).toHaveAttribute('href', `${dataset.FEATURES.url}${value}`);
			});
		});
		describe('And onSearchPage is true', () => {
			let updateOnFilterBadge = jest.fn();
			it('Then Badge Tags/Features should be rendered without links', () => {
				const { rerender } = wrapper;
				rerender(<Dataset {...props} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />);
				props.data.tags.features.map(value => {
					expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
					expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
				});
			});
			it('Then onclick Tags/Features updateOnFilterBadge should be called', () => {
				fireEvent.click(screen.getByTestId(`badge-${props.data.tags.features[0]}`));
				expect(updateOnFilterBadge.mock.calls.length).toBe(1);
				expect(updateOnFilterBadge.mock.calls[0][0]).toEqual(dataset.FEATURES.filter);
				expect(updateOnFilterBadge.mock.calls[0][1]).toEqual({ label: props.data.tags.features[0], parentKey: dataset.FEATURES.parentKey });
			});
		});
	});
	describe('And showRelationshipQuestion is true', () => {
		it('Then the remove button should be rendered ', () => {
			const { rerender } = wrapper;
			rerender(<Dataset {...props} showRelationshipQuestion={true} />);
			expect(screen.getByTestId('closeicon')).toBeTruthy();
		});
		it('Then onclick removeButton function should be called', () => {
			fireEvent.click(screen.getByTestId('closeicon'));
			expect(props.removeButton.mock.calls.length).toBe(1);
		});
		it('Then the description should not be rendered', () => {
			expect(screen.queryByTestId('dataset-description')).toBeNull();
		});
	});

	describe('And abstract is empty', () => {
		it('Then the description button should be rendered', () => {
			const { rerender } = wrapper;
			let data = { ...props.data };
			data.datasetfields.abstract = '';
			rerender(<Dataset {...props} data={data} />);
			expect(screen.getByTestId('dataset-description')).toHaveTextContent(props.data.description);
		});
	});

	describe('And when datasetV2 is empty', () => {
		it('Then Publisher Name should be rendered from datasetfields object in UpperCase', () => {
			const { rerender } = wrapper;
			let data = { ...props.data };
			data.datasetfields.publisher = 'Alliane > TestPublisher';
			delete data.datasetv2;
			data.datasetfields.abstract = '';
			rerender(<Dataset {...props} data={data} />);
			const publisher = screen.getByTestId('publisher-TESTPUBLISHER');
			expect(publisher).toHaveTextContent('TESTPUBLISHER');
		});
		it('Then it should not render sheild Icon', () => {
			expect(screen.queryByTestId(`shield`)).toBeNull();
		});
	});
});
