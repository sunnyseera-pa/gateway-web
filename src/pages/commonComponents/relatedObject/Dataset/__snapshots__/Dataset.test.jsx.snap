// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Dataset component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="noMargin row"
  >
    <div
      class="pad-left-24 col-lg-10 col-sm-10"
    >
      <span
        class="black-bold-16"
        data-testid="title-dataset-c1f4b16c-9dfa-48e5-94ee-f0aa58c270e4"
      >
         
        COVID-19 Symptom Tracker Dataset
      </span>
      <br />
      <span>
        <svg
          class="svg-16 mr-2"
          data-testid="shield"
          height="100%"
          viewBox="0 0 16 16"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M12.3261058,2.07122417 L19.5281873,5.59851616 C19.8057242,5.70704822 20,5.96481187 20,6.26327504 C20,9.92623211 19.5004337,12.7616322 18.3070252,15.1222046 C17.0997398,17.4963434 15.1986123,19.354955 12.39549,21.0779014 C12.1595837,21.2406995 11.8404163,21.2406995 11.60451,21.0779014 C8.80138768,19.354955 6.90026019,17.4963434 5.69297485,15.1222046 C4.49956635,12.7616322 4,9.92623211 4,6.26327504 C4,5.96481187 4.1942758,5.70704822 4.47181266,5.59851616 L11.6738942,2.07122417 C11.8820468,1.97625861 12.1179532,1.97625861 12.3261058,2.07122417 Z M15.7420192,8.13460125 C15.4508655,7.9188202 15.0313953,7.96896159 14.8051062,8.24659519 L14.8051062,8.24659519 L10.5990074,13.4053166 L9.13091008,12.0053922 C8.86906899,11.7642411 8.45285947,11.7676899 8.19545276,12.0131436 C7.93804606,12.2585973 7.93442931,12.6554796 8.18732389,12.9051618 L8.18732389,12.9051618 L10.1892748,14.8141496 C10.3143919,14.9334927 10.4840989,15.0005941 10.6610679,15.0005941 L10.6610679,15.0005941 L10.7024415,15.0005941 C10.8938706,14.9891006 11.0708571,14.8996165 11.1882483,14.754971 L11.1882483,14.754971 L15.859467,9.02800754 C16.0857561,8.75037392 16.0331729,8.35038231 15.7420192,8.13460125 Z"
            fill="#475da7"
          />
        </svg>
      </span>
      <span
        class="gray800-14 css-1oddrib-Dataset"
        data-testid="publisher-SAIL"
      >
         
        SAIL
         
      </span>
    </div>
    <div
      class="pad-right-24 col-lg-2 col-sm-2"
    >
      <div
        class="datasetLogoCircle floatRight css-x0fehg-Dataset"
        data-testid="publisher-logo"
      />
    </div>
    <div
      class="pad-left-24 pad-right-24 pad-top-16 col-lg-12 col-sm-12"
    >
      <div
        class="badge-dataset"
        data-testid="badge-Datasets"
      >
        <svg
          class="badgeSvg mr-2"
          data-testid="dataseticon"
          height="100%"
          viewBox="-2 -2 22 22"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M12,0 C17.5202663,0 22,1.49324457 22,4 L22,4 L22,20 C22,22.5402768 17.5868833,24 12,24 C6.52266339,24 2.17351467,22.5969596 2.00506568,20.1481789 L2,20 L2,4 L2.00514101,3.8537164 C2.17607588,1.43526006 6.58797422,0 12,0 Z M12,16 C8.70211768,16 5.81322554,15.4913704 3.99960013,14.5520997 L4,20 C4,20.8366463 7.5171159,22 12,22 C16.378631,22 19.8358909,20.8901266 19.9943256,20.058882 L20,20 L20.000751,14.5519178 C18.1871473,15.4913048 15.2980952,16 12,16 Z M4,12 C4,12.8366463 7.5171159,14 12,14 C16.378631,14 19.8358909,12.8901266 19.9943256,12.058882 L20,12 L20.0011286,6.52711098 C18.1763026,7.47577408 15.2707986,8 12,8 C8.7298061,8 5.82477173,7.47596789 3.99988365,6.52763709 L4,12 Z M12,2 L11.6929681,2.00191586 C7.41460485,2.05528174 4,3.21181328 4,4 C4,4.80695307 7.5791408,6 12,6 C16.4208592,6 20,4.80695307 20,4 C20,3.19304693 16.4208592,2 12,2 L12,2 Z"
            fill="#113328"
          />
        </svg>
        <span>
          Datasets
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-COVID-19"
      >
        <span>
          COVID-19
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-TRACKER"
      >
        <span>
          TRACKER
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-SAIL"
      >
        <span>
          SAIL
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-COVID"
      >
        <span>
          COVID
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-MOBILE APP"
      >
        <span>
          MOBILE APP
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-CORONAVIRUS"
      >
        <span>
          CORONAVIRUS
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-SYMPTOM"
      >
        <span>
          SYMPTOM
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-NCS"
      >
        <span>
          NCS
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-National Core Study"
      >
        <span>
          National Core Study
        </span>
        <span />
      </div>
    </div>
    <div
      class="pad-left-24 pad-right-24 pad-top-24 pad-bottom-16 col-lg-12 col-sm-12"
    >
      <span
        class="gray800-14"
        data-testid="dataset-description"
      >
        This is a test dataset abstract
      </span>
    </div>
  </div>
</div>
`;
